gui/wxpython/main_window/frame.py

```diff
@@ -961,29 +993,37 @@ def OnCBPageClosing(self, event):
 
         maptree = self.notebookLayers.GetPage(event.GetSelection()).maptree
         maptree.GetMapDisplay().CleanUp()
-        self.mapnotebook.DeletePage(self.GetMapDisplayIndex())
+        if self.GetMapDisplay().IsDocked():
+            self.mapnotebook.DeletePage(self.GetMapDisplayIndex())
+        else:
+            self.GetMapDisplay().GetParent().Destroy()
```

Comment:
I think just `self.GetMapDisplay().Destroy()` should be enough thanks to the FrameMixin.